javascript/app-framework/src/components/app/ResourcesPane/FileTreeCommands.hooks.ts

```diff
@@ -32,29 +40,122 @@ export function useFileManagementCommands(
 	onSaveCommands: IContextualMenuItem[]
 	newCommands: IContextualMenuItem[]
 } {
-	const uploadZip = useUploadZip()
-	const onClickDownloadZip = useDownloadZip()
 	const dataPackage = useDataPackage()
 	const isDataPackageEmpty = useObservableState(
 		dataPackage.isEmpty$,
 		dataPackage.isEmpty,
 	)
+	const newCommands = useNewMenuCommands(plugins)
+	const saveCommands = useSaveMenuCommands(plugins)
+	const openCommands = useOpenMenuItems(examples, plugins, setFile)
+	const theme = useTheme()
+	const buttonStyles = useMemo(
+		() => ({
+			root: {
+				background: theme.palette.neutralLighter,
+			},
+		}),
+		[theme],
+	)
+	const createItem = useCallback(
+		(
+			text: string,
+			iconProps: IIconProps,
+			items: IContextualMenuItem[],
+			extra: Partial<ICommandBarItemProps> = {},
+		): ICommandBarItemProps => {
+			return {
+				key: text,
+				text: expanded ? text : undefined,
+				iconOnly: !expanded,
+				iconProps,
+				buttonStyles,
+				subMenuProps: { items },
+				...extra,
+			}
+		},
+		[buttonStyles, expanded],
+	)
+	const commands = useMemo<ICommandBarItemProps[]>(
+		() => [
+			createItem('New', icons.newFile, newCommands),
+			createItem('Open', icons.openFile, openCommands),
+			createItem('Save', icons.save, saveCommands, {
+				disabled: isDataPackageEmpty,
+			}),
+		],
+		[newCommands, openCommands, saveCommands, isDataPackageEmpty],
+	)
 
-	const onClickUploadTable = useCallback(
-		() =>
-			void onOpenFileRequested(TABLE_TYPES)
-				.then(setFile)
-				.catch(err => console.error('error loading table file', err)),
-		[setFile, onOpenFileRequested],
+	return {
+		commands,
+		newCommands,
+		onOpenCommands: openCommands,
```

Comment:
The "on" prefix here makes this sound like an event hander - use `openCommands` and `saveCommands` to match `newCommands`, which would save the remap too?

Reply:
Agree